tests/native/test_models.py

```diff
@@ -198,3 +225,125 @@ def test_symbolic_mixed(self):
         self.state.constrain(sy[3] != 0)
         ret = strlen(self.state, s)
         self.assertTrue(self.state.must_be_true(ret == 4))
+
+
+class StrcpyTest(ModelTest):
+    def _check_BitVecITE(self, dst, dst_val):
+        # Iterate and check the nested ITE tree for a symbolic byte
+        self.assertTrue(issymbolic(dst))
+        while type(dst.true_value) is BitVecITE:  # check each if/else in dst
+            self.assertEqual(dst.false_value, dst_val)  # dst = false_val
+            dst = dst.true_value
+        return dst
+
+    def _assert_start(self, s, d):
+        # Checks all characters are copied until the 1st that could be NULL
+        cpu = self.state.cpu
+        src = cpu.read_int(s, 8)
+        dst = cpu.read_int(d, 8)
+        offset = 0
+        while cant_be_NULL(src, self.state.constraints):
+            self.assertTrue(not issymbolic(dst))
+            self.assertEqual(src, dst)
+            offset += 1
+            src = cpu.read_int(s + offset, 8)
+            dst = cpu.read_int(d + offset, 8)
+        return offset
+
+    def _assert_end(self, s, d, offset, org_dest_val):
+        cpu = self.state.cpu
+        src = cpu.read_int(s + offset, 8)
+        dst = cpu.read_int(d + offset, 8)
+
+        # src string was entirely symbolic
+        if not issymbolic(dst):
+            self.assertTrue(is_NULL(src, self.state.constraints))
+            self.assertEqual(0, dst)
+
+        # Check each symbolic byte in the dst
+        else:
+            # Loop till src must be NULL or assumed dst space allowed is gone
+            while not is_NULL(src, self.state.constraints) and offset < len(org_dest_val):
+                # Check ITE tree in symbolic dst
+                dst = self._check_BitVecITE(dst, org_dest_val[offset])
+
+                # Check that true value is src
+                if type(dst.true_value) is ArraySelect:
+                    self.assertEqual(type(src), ArraySelect)
+                    self.assertEqual(src.index, dst.true_value.index)
+                    self.assertTrue(src.array is dst.true_value.array)
+                else:
+                    self.assertTrue(Operators.ITE(dst.true_value == src, True, False))
+
+                # Check the false value is dst or offset
+                if issymbolic(src) and can_be_NULL(src, self.state.constraints):
+                    self.assertEqual(dst.false_value, 0)
+                else:
+                    self.assertEqual(dst.false_value, org_dest_val[offset])  # dst = false_val
+
+                offset += 1
+                src = cpu.read_int(s + offset, 8)
+                dst = cpu.read_int(d + offset, 8)
+
+            # Check last sym dst byte
+            dst = self._check_BitVecITE(dst, org_dest_val[offset])
+            self.assertEqual(dst.true_value, 0)
+
+    def _test_strcpy(self, string, dst_len=None):
```

Comment:
Can you add a brief comment (1 or 2 sentences) as to what this method is testing, at a high level?